lib/packetgen/header/http/request.rb

```diff
@@ -85,6 +87,6 @@ def to_s
     end
 
     self.add_class HTTP::Request 
-    TCP.bind_header HTTP::Request, body: ->(b) { /^(CONNECT|DELETE|GET|HEAD|OPTIONS|PATCH|POST|PUT)/ =~ b }
+    TCP.bind_header HTTP::Request, body: ->(b) { /(CONNECT|DELETE|GET|HEAD|OPTIONS|PATCH|POST|PUT)/ =~ b.chars.select(&:valid_encoding?).join }
```

Comment:
Why did you remove the `^` in front of regexp?

Reply:
From what I've seen in some packets: the actual TCP body may not actually start that way, but still be a valid HTTP body to parse.

Now that you mention it: I will go over this again to try and provide a better solution.